pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -69,23 +69,75 @@ type Sensor struct {
 // SensorList is the list of Sensor resources
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type SensorList struct {
-	// +k8s:openapi-gen=false
 	v1.TypeMeta `json:",inline"`
-	// +k8s:openapi-gen=false
 	v1.ListMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
 	Items       []Sensor `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
 // SensorSpec represents desired sensor state
 type SensorSpec struct {
-	// EventDependency is a list of the events that this sensor is dependent on.
-	EventDependencies []EventDependency `json:"dependencies" protobuf:"bytes,1,rep,name=dependencies"`
+	// Dependencies is a list of the events that this sensor is dependent on.
+	Dependencies []EventDependency `json:"dependencies" protobuf:"bytes,1,rep,name=dependencies"`
 
 	// Triggers is a list of the things that this sensor evokes. These are the outputs from this sensor.
 	Triggers []Trigger `json:"triggers" protobuf:"bytes,2,rep,name=triggers"`
 
 	// DeploySpec contains sensor pod specification. For more information, read https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#pod-v1-core
 	DeploySpec *corev1.PodSpec `json:"deploySpec" protobuf:"bytes,3,opt,name=deploySpec"`
+
+	// EventProtocol is the protocol through which sensor receives events from gateway
+	EventProtocol *EventProtocol `json:"eventProtocol" protobuf:"bytes,4,opt,name=eventProtocol"`
+}
+
+// EventProtocol contains configuration necessary to receieve an event from gateway over different communication protocols
+type EventProtocol struct {
+	// Type defines the type of protocol over which events will be receieved
+	Type common.EventProtocolType `json:"type" protobuf:"bytes,1,opt,name=type"`
```

Comment:
okay so you reference this across gateway and sensor APIs.. you should create another api package for the event stuff (even though this package doesn't create any CRDs)